identity-wallet/src/error.rs

```diff
@@ -10,6 +10,8 @@ pub enum AppError {
     // Generic error (all purpose)
     #[error("Error: {0}")]
     Error(&'static str),
+    #[error("Error: {0}")]
+    DevError(String),
```

Comment:
Why the need for a dedicated new type? Why not use the regular generic Error?

Reply:
Is it handled differently?

Reply:
Yes that's because &'static str is already not good because it doesn't allow for dynamic messages (so maybe we should change that as well). This is done because I want to parse an error response of function to a string.

Reply:
You can improve `Error(&'static str)` and use that one then, right? I just wanted to know if there's a specific reason for a dedicated "DevError" since the distinction towards a "regular" Error isn't quite clear to me if it's not handled in a different way.

Reply:
Sure I will do that, but I didn't want to accidentally break something or change something which wasn't discussed.